src/jit/lsra.cpp

```diff
@@ -5671,12 +5686,42 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
         {
             RegRecord* physRegToUnassign = availablePhysRegInterval;
 #ifdef _TARGET_ARM_
-            // We should unassign a double register if availablePhysRegInterval is part of the double register
-            if (availablePhysRegInterval->assignedInterval->registerType == TYP_DOUBLE &&
-                !genIsValidDoubleReg(availablePhysRegInterval->regNum))
-                physRegToUnassign = findAnotherHalfRegRec(availablePhysRegInterval);
+            RegRecord* anotherPhysRegToUnassign  = nullptr;
+            Interval*  anotherIntervalToUnassign = nullptr;
+
+            if (currentInterval->registerType == TYP_DOUBLE)
+            {
+                if (intervalToUnassign->registerType == TYP_FLOAT)
+                {
+                    // We should unassign not only physRegToUnassign which is a float register
+                    // but also another half float register if the currentInterval has a double register.
+                    // So, additional variables would be set.
+                    anotherPhysRegToUnassign = findAnotherHalfRegRec(physRegToUnassign);
+                    assert(anotherPhysRegToUnassign != nullptr);
+                    anotherIntervalToUnassign = anotherPhysRegToUnassign->assignedInterval;
+                    assert(intervalToUnassign != anotherIntervalToUnassign);
+                }
+            }
+            else // TYP_FLOAT
+            {
+
+                // We should unassign a double register if availablePhysRegInterval is part of the double register
+                if (intervalToUnassign->registerType == TYP_DOUBLE &&
+                    !genIsValidDoubleReg(availablePhysRegInterval->regNum))
+                {
+                    physRegToUnassign = findAnotherHalfRegRec(availablePhysRegInterval);
+                }
+            }
 #endif
             unassignPhysReg(physRegToUnassign, intervalToUnassign->recentRefPosition);
```

Comment:
@CarolEidt @hqueue Thank you guys. Let me check what you guys said and change again this.